### PR TITLE
split sticky / non sticky post, register implementation

### PR DESCRIPTION
**Pull request by dmitryPoznukhov**

### CODE REVIEW

File: Swen/Classes/Swen.swift [hunk 1]
```diff
@@ -8,15 +8,11 @@
 
 import Foundation
 
-public protocol Event {
+public protocol EventBase {}
```
Comment (carmen-gh) [inline Swen/Classes/Swen.swift]: why did you add EventBase? what is the difference of EventBase and Event?
EventBase sounds a little bit odd

Comment (dmitryPoznukhov) [reply Swen/Classes/Swen.swift]: may be it make sense to rename to BaseEvent, main idea - to achieve 
```
public extension Swen where EventType: Event {
and
public extension Swen where EventType: StickyEvent {
```

File: Swen/Classes/Swen.swift [hunk 2]
```diff
@@ -86,18 +118,25 @@ fileprivate extension Swen {
         listeners.append(listener)
         if let sticky = sticky {
             listener.queue.addOperation {
-                listener.post(sticky)
+                handler(sticky)
```
Comment (mobilsixt) [inline Swen/Classes/Swen.swift]: Why are you not calling listener.post instead of queue.addOperation?

Comment (FranzBusch) [inline Swen/Classes/Swen.swift]: Why not just call listener.post ?

Comment (dmitryPoznukhov) [reply Swen/Classes/Swen.swift]: in this way we do force dispatch, in case listener.post, if this called from the same queue as listener.queue we will have deadlock

Comment (FranzBusch) [reply Swen/Classes/Swen.swift]: Why would there be a deadlock if it is the same thread?
Let's say both are on the main queue:
Register->Listener.init()->listener.post()->handler()

Where would the deadlock occur?

Comment (dmitryPoznukhov) [reply Swen/Classes/Swen.swift]: ah, yes, true, it just skip OperationQueue check

Comment (FranzBusch) [reply Swen/Classes/Swen.swift]: So we can just call listener.post ?
If so you can change that and i'll approve :)
